lib/auto-languageclient.js

```diff
@@ -82,6 +82,12 @@ export default class AutoLanguageClient {
     return this.getGrammarScopes().includes(editor.getGrammar().scopeName);
   }
 
+  // Determine whether we should automatically restart a server that exited or
+  // crashed unexpectedly
+  shouldRestartFailedServer(server: ActiveServer): Promise<boolean> {
+    return Promise.resolve(true);
```

Comment:
For now, default behavior is to always restart an exited server.  Should I add a simple check to see if the server has restarted more than X times in Y seconds and then return false if that threshold is reached?

Reply:
Ideally yeah, I think the VSCode one restarts a maximum of 5 times in 3 minutes before reporting errors.  

We should also add some information about the number of restarts/times to the api if we go down that route.

I'd recommend we just auto-restart with the 5 times/3 minutes internally for now and open up the api in a future release if it's needed and we know more about what information is useful in determining whether to restart (e.g. cause, exit codes etc)